test/load/allocation/grpc/runscenario.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2022 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	pb "agones.dev/agones/pkg/allocation/go"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type allocErrorCode string
+
+const (
+	noerror                   allocErrorCode = "Noerror"
+	unknown                   allocErrorCode = "Unknown"
+	objectHasBeenModified     allocErrorCode = "ObjectHasBeenModified"
+	tooManyConcurrentRequests allocErrorCode = "TooManyConcurrentRequests"
+	noAvailableGameServer     allocErrorCode = "NoAvailableGameServer"
+)
+
+var (
+	logger    = log.New(os.Stdout, "", 0)
+	errorMsgs = map[allocErrorCode]string{
+		objectHasBeenModified:     "the object has been modified",
+		tooManyConcurrentRequests: "too many concurrent requests",
+		noAvailableGameServer:     "no available GameServer to allocate",
+	}
```

Comment:
Maybe we can add these errors as well which can be helpful:

		connectionResetByPeer:               "read: connection reset by peer",
		connectionTermination:               "reset reason: connection termination",

Reply:
I didn't see either of those during my test run. I figured I'd only include errors that I had observed and if unknown errors crop up during future runs we can add them in as well.